overlay.d/05rhcos/usr/libexec/rhcos-upgrade-root-filesystem-container

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+set -euxo pipefail
+
+if [ -f /run/device_mod ]; then
+    exit
+fi
+
+# check for the existence of the dm-linear mapping
+if [ -h "/dev/mapper/coreos-luks-root-nocrypt" ]; then
+    dm_device=$(readlink /dev/mapper/coreos-luks-root-nocrypt -f)
+    dm_device_name=$(basename $dm_device)
+    partition=$(ls /sys/block/$dm_device_name/slaves/)
+    parent=$(basename "$(lsblk --noheadings --output PKNAME --paths /dev/$partition | head -n1)")
+    partnum=$(cat /sys/block/$parent/$partition/partition)
+    # hard-code the label to root
+    label="root"
+    guid=$(lsblk "/dev/$partition" -no PARTUUID)
+    # move the start sector 32768 to move past the LUKS header
+    startsector="$(($(cat /sys/block/$parent/$partition/start)+32768))"
+    # the dm device is already the correct size
+    size=$(cat /sys/block/$dm_device_name/size)
+
+    # check / handle 4k sector size devices
+    sector_size=$(cat /sys/block/$parent/queue/logical_block_size)
+    if [ "$sector_size" == "4096" ]; then
+        startsector=$((startsector / 8))
+        size=$((size / 8))
+    fi
+
+    # validate that the new partition will start at the xfs super block
+    skipcount=$(($startsector * $sector_size))
+    if [ $(dd if=/dev/$parent bs=1 count=4 status=none skip=$skipcount) != XFSB ]; then
+        echo "couldn't detect XFS superblock at expected partition start"
+        exit
+    fi
+
+    # recreate the partition offset after the LUKS header
+    # we don't want to copy the typeguid because that's pointing to crypt_luks
+    sgdisk -d $partnum -n $partnum:$startsector:+$size -c $partnum:$label /dev/$parent -u $partnum:$guid
+    # delete the rhcos.root karg to prevent unlocking attempts
+    rpm-ostree kargs --delete rhcos.root
+
+    > /run/device_mod
+    exit
+fi
+
+# pre-4.7 root LUKS devices are always named luks-00000000-0000-4000-a000-000000000002
+if [ -h /dev/mapper/luks-00000000-0000-4000-a000-000000000002 ]; then
```

Comment:
Hmm, feels like this should be an `else` instead, no? I.e. either we're handling the encrypted case, or the non-encrypted case. 

Then we can factor out the `> /run/device_mod` stamp file and drop the explicit `exit`s.

Reply:
I agree that this could be an `elif` but I think we still want the stamp file. Both cases will technically not do anything but we'd fail the XFS superblock check in the case of nocrypt and in the crypt case we'd exit on `rhcos.root` being missing (if you `systemctl restart rhcos-upgrade-root-filesystem-container` during the same first boot the unit will still be re-ran because we've only staged the kargs update with ostree).